File: src/main/java/io/github/rmuhamedgaliev/arcana/core/Language.java
package io.github.rmuhamedgaliev.arcana.core;

/**
 * Enum representing supported languages in the game.
 */
public enum Language {
    EN("English"),
    RU("Русский");

    private final String displayName;

    Language(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    @Override
    public String toString() {
        return displayName;
    }
}

File: src/main/kotlin/io/github/rmuhamedgaliev/arcana/domain/model/Language.kt
package io.github.rmuhamedgaliev.arcana.domain.model

/**
 * Enum representing supported languages.
 */
enum class Language(val code: String, val displayName: String) {
    EN("en", "English"),
    RU("ru", "Русский");

    companion object {
        /**
         * Get a language by its code.
         *
         * @param code The language code
         * @return The language, or null if not found
         */
        fun fromCode(code: String): Language? {
            return values().find { it.code.equals(code, ignoreCase = true) }
        }
    }
}
